go-interface-tutorial/main_test.go

```diff
@@ -101,3 +107,127 @@ func addBookWithTitle(title string) {
 
 	p.sdb.Query("INSERT INTO books VALUES ($1)", title)
 }
+
+func TestHandlerWithNoQuery(t *testing.T) {
+	connStr := "host=localhost port=5400 user=docker password=docker sslmode=disable"
+	db, err := sql.Open("postgres", connStr)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	req, err := http.NewRequest("GET", "/", nil)
+	if err != nil {
+		return
+	}
+	w := httptest.NewRecorder()
+	api := API{
+		repository: PostgresRepository{
+			sdb: &ShopDB{db},
+		},
+	}
+	api.searchHandler(w, req)
+
+	resp := w.Result()
+	body, _ := ioutil.ReadAll(resp.Body)
+	fmt.Println(string(body))
+
+	if resp.Status != "400 Bad Request" {
```

Comment:
Can you also check that the response body contains the same message you write in the handler?